managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1462,7 +1458,19 @@ public void asyncDelete(Position pos, final AsyncCallbacks.DeleteCallback callba
                     newMarkDeletePosition = range.upperEndpoint();
                 }
             }
+        } catch (Exception e) {
```

Comment:
Yes, I think this was the source of the deadlock, I had it fixed in my branch as well. ecf450173d170a9324425c94f0480b8d3feb8c90